Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -190,42 +191,53 @@ struct KlaviyoState: Equatable, Codable {
-            case .organization:
-                attributes.organization = attributes.organization ?? value.value as? String
-            case .city:
-                location.city = location.city ?? value.value as? String
-            case .region:
-                location.region = location.region ?? value.value as? String
-            case .country:
-                location.country = location.country ?? value.value as? String
-            case .zip:
-                location.zip = location.zip ?? value.value as? String
-            case .image:
-                attributes.image = attributes.image ?? value.value as? String
-            case .latitude:
-                location.latitude = location.latitude ?? value.value as? Double
-            case .longitude:
-                location.longitude = location.longitude ?? value.value as? Double
-            case let .custom(customKey: customKey):
-                properties[customKey] = properties[customKey] ?? value.value
-            }
+        if let firstName = updatedProfile.firstName {
+            attributes.firstName = attributes.firstName ?? firstName
         }
+        if let lastName = updatedProfile.lastName {
+            attributes.lastName = attributes.lastName ?? lastName
+        }
+        if let title = updatedProfile.title {
+            attributes.title = attributes.title ?? title
+        }
+        if let organization = updatedProfile.organization {
+            attributes.organization = attributes.organization ?? organization
+        }
+        if !updatedProfile.properties.isEmpty {
+            attributes.properties = AnyCodable(properties.merging(updatedProfile.properties, uniquingKeysWith: { _, new in new }))
+        }
+
+        if let address1 = updatedProfile.location?.address1 {
+            location.address1 = location.address1 ?? address1
+        }
+        if let address2 = updatedProfile.location?.address2 {
+            location.address2 = location.address2 ?? address2
+        }
+        if let city = updatedProfile.location?.city {
+            location.city = location.city ?? city
+        }
+        if let region = updatedProfile.location?.region {
+            location.region = location.region ?? region
+        }
+        if let country = updatedProfile.location?.country {
+            location.country = location.country ?? country
+        }
+        if let zip = updatedProfile.location?.zip {
+            location.zip = location.zip ?? zip
+        }
+        if let image = updatedProfile.image {
+            attributes.image = attributes.image ?? image
+        }
+        if let latitude = updatedProfile.location?.latitude {
+            location.latitude = location.latitude ?? latitude
+        }
+        if let longitude = updatedProfile.location?.longitude {
+            location.longitude = location.longitude ?? longitude
+        }
+
         attributes.location = location
-        attributes.properties = AnyCodable(properties)
```

Comment:
Merging the two dictionaries now.

`attributes.properties = AnyCodable(properties.merging(updatedProfile.properties, uniquingKeysWith: { _, new in new }))`